1_fetch/src/get_nwis_data.R

```diff
@@ -1,48 +1,91 @@
-
-download_nwis_data <- function(site_nums = c("01427207", "01432160", "01435000", "01436690", "01466500")){
-  
-  # create the file names that are needed for download_nwis_site_data
-  # tempdir() creates a temporary directory that is wiped out when you start a new R session; 
-  # replace tempdir() with "1_fetch/out" or another desired folder if you want to retain the download
-  download_files <- file.path(tempdir(), paste0('nwis_', site_nums, '_data.csv'))
-  data_out <- data.frame()
-  # loop through files to download 
-  for (download_file in download_files){
-    download_nwis_site_data(download_file, parameterCd = '00010')
-    # read the downloaded data and append it to the existing data.frame
-    these_data <- read_csv(download_file, col_types = 'ccTdcc')
-    data_out <- bind_rows(data_out, these_data)
-  }
-  return(data_out)
+download_nwis_data <- function(site_num, out_dir, startDate, endDate, parameterCd){
+  
+  # function to simultaneously download site information and data for set parameter
+  
+    # create the file name needed for download_nwis_site_data
+      download_file <- file.path(out_dir, paste0("nwis_", site_num, "_nwisdata.csv"))
+        
+    # download parameter data
+      val <- retry(download_nwis_site_data(download_file, startDate, endDate, parameterCd = parameterCd), maxErrors = 10, sleep = 2)
+        
+    # download site info
+      val <- retry(download_nwis_site_info(site_num, out_dir), maxErrors = 10, sleep = 2)
+        
+      return(paste0(out_dir, "nwis_", site_num, "_nwisdata.csv"))
```

Comment:
You really only want each function to create one _thing_ when using a pipeline. In the way you have this function written, it is pulling both data and the site metadata, but because you can only return the one file path, your pipeline will only be tracking the data file, not the site metadata file.

I think the easiest way to tackle this is to continue with downloading each individual site's data as separate targets (as you have already implemented), then combine them, and then use the combined data to pull site info for the unique sites in that data and save as one file.

Reply:
Makes sense! I addressed this and your other comments here in the latest push.